experimental/sys/errno.go

```diff
@@ -0,0 +1,95 @@
+package sys
+
+import "strconv"
+
+// Errno is a subset of POSIX errno used by wazero interfaces. Zero is not an
+// error. Other values should not be interpreted numerically, rather by constants
+// prefixed with 'E'.
+//
+// See https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/errno.h.html
+type Errno uint16
+
+// ^-- Note: This will eventually move to the public /sys package. It is
```

Comment:
Here's the note on why, but TL;DR; we already have an internal/sys package but we need one like the public sys one (shared symbols regardless of filesystem, socket, proc, etc.) By placing this into experimental, we can use the same package name and in a lot of cases refer to these errors as `sys.Errno` and especially in case of docs, this name will prevent drift mistakes, as eventually it will be the public package `sys`